chingu-voyages/v52-tier3-team-38
# Gary/fix/logout and rendering

# Description
This PR fixes the following issues. Logout functionality is restored, GoogleOauth now fully functions as it was supposed to, Home page renders immediately after logins/logouts. Some refactors were made to handle removal of the user_details table.

## Related to the following Jira/Trello issue 
N/A

## Main changes explained:
- AdminHeader and UserHeader now uses logout function from actions.js to correctly handle logout.
- Converted page.js ('/') to be a server component to correctly show info pending on who is logged in.
- Adjusted LoginInWithGoogle function to include api to finish google authentication, providing session information about the admin.
- Signup form now only needs name, email, and password.

## How to test:
1. check into current branch
2. do `npm install` and `npm run dev` to run this PR locally
3. Sign up for application as a user, when signed up you should see be redirected to home page where user page is present, logging out should show guest page.
4. Login as an admin, the home page should render as the admin page, test log out functionality again. Should once again see guest page when logged out.
